priv/migration_tasks/prod/1.5.14@add_geopatch_node_transactiosn.ex

```diff
@@ -0,0 +1,159 @@
+defmodule Migration_1_5_14 do
+  @moduledoc """
+  Migration script to add geopatch to a node's transaction.
+  """
+
+  alias Archethic.Crypto
+  alias Archethic.P2P
+  alias Archethic.P2P.Node
+  alias Archethic.TransactionChain
+  alias Archethic.TransactionChain.Transaction
+  alias Archethic.TransactionChain.TransactionData
+  alias Archethic.Utils
+  alias Archethic.PubSub
+
+  require Logger
+
+  def run() do
+    nodes = P2P.list_nodes() |> Enum.sort_by(& &1.first_public_key)
+
+    execute_migration(nodes)
+  end
+
+  defp execute_migration([]), do: :ok
+
+  defp execute_migration(nodes) do
+    current_node_pk = Crypto.first_node_public_key()
+
+    Enum.reduce_while(nodes, :ok, fn node, _acc ->
+      node_pk = node.first_public_key
+
+      if node_pk == current_node_pk do
+        Logger.info("Starting migration for: #{Base.encode16(node_pk)}")
+
+        case send_node_transaction() do
+          :ok ->
+            Logger.info("Migration complete for: #{Base.encode16(node_pk)}")
+            {:halt, :ok}
+
+          {:error, reason} ->
+            Logger.error("Migration failed for: #{Base.encode16(node_pk)}")
+            {:halt, {:error, reason}}
+        end
+      else
+        PubSub.register_to_new_transaction_by_type(:node)
+
+        receive do
+          {:new_transaction, address, :node, _timestamp} ->
+            with {:ok, %Transaction{previous_public_key: previous_pk} = transaction} <-
+                   TransactionChain.get_transaction(address) do
+              first_pk = TransactionChain.get_first_public_key(previous_pk)
+
+              if first_pk == node_pk do
```

Comment:
I think if you receive a transaction from an other node than expected you should have a specific behavior. For example if you are waiting for Node1 transaction and you receive Node2 transaction, you will continue in the list and then wait for Node2 transaction, but you just received it